v2/oauth/multiple-frontends-with-a-single-backend.mdx

```diff
@@ -0,0 +1,288 @@
+- `redirect_uri` - A list of URIs to which the [**Authorization Server**](/docs/oauth/introduction#authorization-server) will send the user-agent (browser) after completing the authorization step. These can be deep links to mobile or desktop apps as well, but they must be exact URLs, without wildcards.
+
+If the creation was successful, the API will return a response that looks like this:
+
+```json
+
+{
+    "client_name": "<YOUR_CLIENT_NAME>",
+    "client_id": "<CLIENT_ID>",
+    "client_secret": "<CLIENT_SECRET>",
+    "callback_urls": ["https://<YOUR_APPLICATION_DOMAIN>/oauth/callback"],
+    "authorize_url": "^{form_apiDomain}/^{form_apiBasePath}/oauth/authorize",
+    "token_fetch_url": "^{form_apiDomain}/^{form_apiBasePath}/oauth/token"
+}
+
+```
+
+:::caution
+
+You will have to save this response because we do not persist it interaly for security reasons.
+In the next steps we will use the values to complete several configurations.
+
+:::
+
+
+## 3. Set Up your Authorization Service 
+
+
+### Configure the Authorization Service Backend
+
+In your [**Authorization Service**](/docs/oauth/introduction#authorization-service) you will need to initialize the **OAuth2Provider** and the **OAuth2Client** recipes.
+
+<BackendSDKTabs  disableOtherFrameworks>
+<TabItem value="nodejs">
+
+Update the `supertokens.init` call to include the new recipes.
+
+1. Add the new import statements 
+
+```typescript 
+import OAuth2Provider from "supertokens-node/recipe/oauth2provider";
+import OAuth2Client from "supertokens-node/recipe/oauth2client";
+```
+
+2. Update the list of recipes with the new initialization steps.
+
+```tsx
+OAuth2Provider.init(),
+OAuth2Client.init({
+  // Use the values returned from the previous client creation step
+  providerConfigs: [{
+    "clientId": "<YOUR_CLIENT_ID>",
+    "clientSecret": "<YOUR_CLIENT_SECRET>",
+  }]
+})
+```
+</TabItem>
+
+<TabItem value="go">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Go SDK.
+
+:::
+
+</TabItem>
+
+<TabItem value="python">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Python SDK.
+
+:::
+
+</TabItem>
+
+</BackendSDKTabs>
+
+###  Configure the Authorization Service Frontend 
+
+Initialize the `OAuth2Provider` recipe on the frontend of your **Authorization Service**. 
+
+:::info
+
+If you want to use your own custom UI check our [separate guide](/docs/oauth/customizations/custom-ui) that explains
+all the steps that you have to take into account.
+
+:::
+
+<OAuthFrontendTabs>
+
+
+<TabItem value="react">
+
+Update the `SuperTokensConfig` variable to adjust to the new flow. 
+
+1. Add the import statement for the recipe.
+
+```typescript 
+import OAuth2Provider from "supertokens-auth-react/recipe/oauth2provider";
+```
+
+2. Update the list of recipes to also include the new initialization.
+
+```tsx
+OAuth2Provider.init()
+```
+
+3. Update the **Session** recipe initialization to use `header` authentication.
```

Comment:
this is not needed to do in the authorization server frontend. This step is needed in the application frontend

Reply:
resolved